utt/tests/testUtils.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+#include "details.hpp"
+#include "bankIdentity.hpp"
+#include "clientIdentity.hpp"
+#include "registratorIdentity.hpp"
+#include "common.hpp"
+#include "coin.hpp"
+#include <utt/RegAuth.h>
+#include <utt/RandSigDKG.h>
+#include <utt/Serialization.h>
+#include <utt/Params.h>
+#include <utt/IBE.h>
+#include <utt/Serialization.h>
+#include <utt/Address.h>
+#include <utt/Comm.h>
+
+#include <vector>
+#include <memory>
+#include <cstdlib>
+#include <iostream>
+#include <ctime>
+using namespace libutt;
+using namespace libutt::api;
+namespace libutt::api::testing {
+std::vector<uint32_t> getSubGroup(uint32_t n, uint32_t size) {
+  std::srand((unsigned int)std::time(0));
```

Comment:
This is only for testing, so it is not that important IMO

Reply:
I would agree with you if this is a temp function, but if it's going to be used in tests going forward I would say it's important. To me it seems that the seeding here is done in case someone forgets to seed before using getSubGroup. Is there any advantage to seed inside the function besides that? IMO you can seed once in each test and name the function "getReandomSubGroup" to hint that it does use randomness. And btw, there are other ways to implement "sampling without replacement", including some new stuff from C++17 (https://stackoverflow.com/questions/33802205/how-to-sample-without-replacement-using-c-uniform-int-distribution) - see the second post.